client/src/main/java/io/pravega/client/BatchClientFactory.java

```diff
@@ -80,6 +81,22 @@ static BatchClientFactory withScope(String scope, ClientConfig config) {
      */
     <T> SegmentIterator<T> readSegment(SegmentRange segment, Serializer<T> deserializer);
 
+    /**
+     * Return first offset for a given {@link Segment}.
+     *
+     * @param segment The {@link Segment}
+     * @return {@link StreamCut} pointing to the starting offset of the {@link Segment}.
+     */
+    StreamCut firstStreamCut(Segment segment);
+
+    /**
+     * Return last offset for a given {@link Segment}.
+     *
+     * @param segment The {@link Segment}
+     * @return {@link StreamCut} pointing to the last offset of the {@link Segment}.
+     */
+    StreamCut lastStreamCut(Segment segment);
+
```

Comment:
These are returning a StreamCut object. But because the data within the StreamCut only pertains to one segment, it isn't a well formed StreamCut.
I think what you want to do is have a method:
```
SegmentRange currentSegmentRange(Segment segment);
```
The implementation of which would grab the lowest and highest offsets from the segment is the implementation of the two methods here do, and return it as a SegmentRange.

Reply:
thanks a lot!  so, i made the changes.  i still need to verify the changes on my app side (changing repos, build systems, so a bit delayed on that there), and i'll change from draft after that